### PR TITLE
[ACS-3960][ACS-4339] Count is only returned for a tag if it is greater than zero

### PR DESCRIPTION
**Pull request by pjoshi31**
[ACS-3960][ACS-4339] Count is only returned for a tag if it is greater than zero

[ACS-3960]: https://alfresco.atlassian.net/browse/ACS-3960?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java [hunk 1]
```diff
@@ -179,7 +179,7 @@ public CollectionWithPagingInfo<Tag> getTags(StoreRef storeRef, Parameters param
         for (Pair<NodeRef, String> pair : page)
         {
             Tag selectedTag = new Tag(pair.getFirst(), pair.getSecond());
-            selectedTag.setCount(tagsByCountMap.get(selectedTag.getTag()));
+            selectedTag.setCount(tagsByCountMap.get(selectedTag.getTag()) == null ? 0 : tagsByCountMap.get(selectedTag.getTag()));
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java:182]:
```suggestion
            selectedTag.setCount(Optional.ofNullable(tagsByCountMap.get(selectedTag.getTag()).orElse(0));
```

**Review by tpage-alfresco — APPROVED** [review]
LGTM

**Comment by tpage-alfresco** [comment]
It would be good to have a unit test too.

File: remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java [hunk 2]
```diff
@@ -83,6 +84,21 @@ public void setup()
...
+        given(taggingServiceMock.createTags(any(), any())).willAnswer(invocation -> createTagAndNodeRefPairs(invocation.getArgument(1)));
+        given(parametersMock.getInclude()).willReturn(List.of("count"));
+        final List<Tag> actualCreatedTags = objectUnderTest.createTags(tagsToCreate, parametersMock);
+        final List<Tag> expectedTags = createTagsWithNodeRefs(tagNames).stream()
+            .peek(tag -> tag.setCount(0))
+            .collect(Collectors.toList());
+        assertThat(actualCreatedTags)
+            .isNotNull()
+            .isEqualTo(expectedTags);
+        assertEquals(expectedTags, actualCreatedTags);
```
Comment (tpage-alfresco) [inline remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java:100]: I think you can reduce these four lines down to one. The final assertion covers everything above it.

Comment (pjoshi31) [reply remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java:97]: Incorporated

**Pull request merged**